src/lib/quoter.js

```diff
@@ -1,17 +1,100 @@
 'use strict'
 
 const ILQP = require('ilp').ILQP
+const LiquidityCurve = require('ilp-routing').LiquidityCurve
+const IlpPacket = require('ilp-packet')
 const InvalidAmountSpecifiedError = require('../errors/invalid-amount-specified-error')
 const IlpError = require('../errors/ilp-error')
 
 class Quoter {
-  constructor (ledgers) {
+  /**
+   * @param {Ledgers} ledgers
+   * @param {Object} config
+   * @param {Integer} config.quoteExpiry
+   */
+  constructor (ledgers, config) {
     this.ledgers = ledgers
     this.tables = ledgers.tables
+    this.localTables = this.tables.localTables
+    this.quoteExpiryDuration = config.quoteExpiry // milliseconds
+  }
+
+  /**
+   * @param {Object} request
+   * @param {IlpAddress} request.sourceAccount
+   * @param {IlpAddress} request.destinationAccount
+   * @param {Integer} request.destinationHoldDuration
+   * @param {Boolean} [request._shiftCurve] default: true
+   * @returns {Object}
+   */
+  * quoteLiquidity (request) {
+    const liquidityQuote = yield this._quoteLiquidity(
+      Object.assign({_shiftCurve: true}, request))
+    if (!liquidityQuote) return null
+    return Object.assign({}, liquidityQuote, {
+      liquidityCurve: liquidityQuote.liquidityCurve.toBuffer(),
+      expiresAt: new Date(liquidityQuote.expiresAt)
+    })
+  }
+
+  * _quoteLiquidity (request) {
+    const hop = this.localTables.findBestHopForSourceAmount(
+      request.sourceAccount, request.destinationAccount, '0')
+    if (!hop) return Promise.resolve(null)
+    const connector = hop.bestHop
+    let fullRoute = hop.bestRoute
+    if (isCurveExpired(fullRoute)) {
+      const tailQuote = yield ILQP.quoteByConnector({
+        plugin: this.ledgers.getPlugin(fullRoute.nextLedger),
+        connector,
+        quoteQuery: {
+          destinationAccount: request.destinationAccount,
+          destinationHoldDuration: request.destinationHoldDuration
+        }
+      })
+      if (tailQuote.responseType === IlpPacket.Type.TYPE_ILP_ERROR) {
+        throw new IlpError(tailQuote)
+      }
+      const tailCurve = new LiquidityCurve(tailQuote.liquidityCurve)
+      // The quote is more specific than the route.
+      if (fullRoute.targetPrefix.length < tailQuote.appliesToPrefix.length) {
+        this.localTables.addRoute({
+          source_ledger: fullRoute.nextLedger,
+          destination_ledger: tailQuote.appliesToPrefix,
+          source_account: connector,
+          min_message_window: (tailQuote.sourceHoldDuration - request.destinationHoldDuration) / 1000,
+          points: tailQuote.liquidityCurve
+        })
+        fullRoute = this.localTables.findBestHopForSourceAmount(
+          request.sourceAccount, request.destinationAccount, '0').bestRoute
```

Comment:
Yes, I think so. It is accessing the route `request.sourceAccount -> request.destinationAccount`, immediately after creating a route that will match that specification.